### PR TITLE
Fix pinhole offset corrections

### PR DESCRIPTION
**Pull request by saransh13**
fix pinhole corrections for edge cases. All variables for PXRDIP and TARDIS should be same now. No special treatment of PXRDIP data needed.

### CODE REVIEW

**Comment by saransh13** [comment]
@psavery This should solve our problem of having different eta angles for `PXRDIP` and `TARDIS` in the `rygg pinhole correction` functions. Can you please review?

File: hexrd/xrdutil/phutil.py [hunk 1]
```diff
@@ -449,7 +452,7 @@ def _infer_eHat_l(panel):
 
     eHat_l_dict = {
         'TARDIS': -ct.lab_x.reshape((3, 1)),
-        'PXRDIP': ct.lab_y.reshape((3, 1))
+        'PXRDIP': -ct.lab_x.reshape((3, 1))
```
Comment (psavery) [inline hexrd/xrdutil/phutil.py:455]: Maybe we can get rid of this function entirely if `PXRDIP` and `TARDIS` are the same.

File: hexrd/xrdutil/phutil.py [hunk 2]
```diff
@@ -460,7 +463,7 @@ def _infer_eta_shift(panel):
 
     eta_shift_dict = {
         'TARDIS': -np.radians(180),
-        'PXRDIP': -np.radians(90),
+        'PXRDIP': -np.radians(180),
```
Comment (psavery) [inline hexrd/xrdutil/phutil.py:466]: This one too. Maybe we can get rid of it also.

File: hexrd/xrdutil/phutil.py [hunk 3]
```diff
@@ -392,8 +392,11 @@ def calc_phi_x(bvec, eHat_l):
     """
     bv = np.array(bvec)
     bv[2] = 0.
-    bv = bv/np.linalg.norm(bv)
-    return np.arccos(np.dot(bv, -eHat_l)).item()
+    if np.linalg.norm(bv) == 0.:
```
[the remark this answers is not in the document]
Comment (saransh13) [reply hexrd/xrdutil/phutil.py:395]: I agree. I will accept your suggestions.

File: hexrd/xrdutil/phutil.py (hunk 1 above)
Comment (saransh13) [reply hexrd/xrdutil/phutil.py:456]: I think the FIDDLE instrument might have to be treated separately. I'm not 100% about that yet, but it's possible. So I'd keep the cases for now. We can remove in a future PR if the FIDDLE instrument doesn't need a separate case.

File: hexrd/xrdutil/phutil.py (hunk 2 above)
Comment (saransh13) [reply hexrd/xrdutil/phutil.py:467]: Same as above.

File: hexrd/xrdutil/phutil.py [hunk 4]
```diff
@@ -392,8 +392,12 @@ def calc_phi_x(bvec, eHat_l):
     """
     bv = np.array(bvec)
     bv[2] = 0.
-    bv = bv/np.linalg.norm(bv)
-    return np.arccos(np.dot(bv, -eHat_l)).item()
+    bv_norm = np.linalg.norm(bv)
+    if np.isclose(bv_norm, 0):
+        return 0.
+    else:
+        bv = bv/np.linalg.norm(bv)
```
Comment (psavery) [inline hexrd/xrdutil/phutil.py:399]:
```suggestion
        bv = bv / bv_norm
```

One last change.

**Review by psavery — APPROVED** [review]
I had one last change requested, but I'll go ahead and approve too.

**Pull request merged**